lib/puppet/network/http/api/indirected_routes.rb

```diff
@@ -192,7 +192,12 @@ def read_body_into_model(model_class, request)
     data = request.body.to_s
 
     format = request.format
-    model_class.convert_from(format, data)
+    begin
+      model_class.convert_from(format, data)
+    rescue => e
+      raise Puppet::Network::HTTP::Error::HTTPBadRequestError.new(
```

Comment:
Unknown Content-Type should be a 415 per https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.16